torch/nn/_functions/thnn/sparse.py

```diff
@@ -103,3 +103,152 @@ def backward(self, grad_output):
 
 
 _all_functions.append(Embedding)
+
+
+class EmbeddingBag(Function):
+
+    def __init__(self, max_norm, norm_type, scale_grad_by_freq, mode):
+        super(EmbeddingBag, self).__init__()
+        self.max_norm = max_norm
+        self.norm_type = norm_type
+        self.scale_grad_by_freq = scale_grad_by_freq
+        self._indices = None
+        assert mode is not None
+        if mode == 'sum':
+            self.mode = 0
+        elif mode == 'mean':
+            self.mode = 1
```

Comment:
can we please not do this? I can't see any value in converting readable mode names to numbers that don't have any meaning. If you want to do this conversion only once, define them as constants:
```python
MODE_SUM = 0
MODE_MEAN = 1
```